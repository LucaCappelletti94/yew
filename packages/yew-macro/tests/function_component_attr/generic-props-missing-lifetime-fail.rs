use yew::prelude::*;

#[derive(Clone, Properties, PartialEq)]
struct Props {
    a: usize,
}

#[function_component(Comp)]
fn comp<P>(_props: &P) -> Html
where
    P: Properties + PartialEq,
{
    html! {
        <p></p>
    }
}

struct MissingTypeBounds;

fn compile_fail_missing_lifetime() {
    // missing prop 'a'
    html! { <Comp<Props> /> };
}

fn main() {}
