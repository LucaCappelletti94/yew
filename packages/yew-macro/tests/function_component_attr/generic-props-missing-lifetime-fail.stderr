error[E0277]: the trait bound `AssertAllProps: HasProp<a, _>` is not satisfied
  --> tests/function_component_attr/generic-props-missing-lifetime-fail.rs:22:14
   |
22 |     html! { <Comp<Props> /> };
   |              ^^^^ the trait `HasProp<a, _>` is not implemented for `AssertAllProps`
   |
   = help: the following other types implement trait `HasProp<P, How>`:
             <HasPropsa<B> as HasProp<a, HasPropsa<B>>>
             <HasPropsa<B> as HasProp<P, &dyn HasProp<P, How>>>
             <CheckPropsAll<B> as HasProp<P, &dyn HasProp<P, How>>>
             <HasContextProviderPropscontext<B> as HasProp<yew::context::_ContextProviderProps::context, HasContextProviderPropscontext<B>>>
             <HasContextProviderPropscontext<B> as HasProp<P, &dyn HasProp<P, How>>>
             <HasContextProviderPropschildren<B> as HasProp<children, HasContextProviderPropschildren<B>>>
             <HasContextProviderPropschildren<B> as HasProp<P, &dyn HasProp<P, How>>>
             <CheckContextProviderPropsAll<B> as HasProp<P, &dyn HasProp<P, How>>>
           and $N others
note: required for `CheckPropsAll<AssertAllProps>` to implement `HasAllProps<Props, (_,)>`
  --> tests/function_component_attr/generic-props-missing-lifetime-fail.rs:3:17
   |
3  | #[derive(Clone, Properties, PartialEq)]
   |                 ^^^^^^^^^^ unsatisfied trait bound introduced in this `derive` macro
   = note: required for `AssertAllProps` to implement `AllPropsFor<PropsBuilder, (_,)>`
note: required by a bound in `yew::html::component::properties::__macro::PreBuild::<Token, B>::build`
  --> $WORKSPACE/packages/yew/src/html/component/properties.rs
   |
   |         pub fn build<How>(self) -> B::Output
   |                ----- required by a bound in this associated function
   |         where
   |             Token: AllPropsFor<B, How>,
   |                    ^^^^^^^^^^^^^^^^^^^ required by this bound in `PreBuild::<Token, B>::build`
   = note: this error originates in the macro `html` which comes from the expansion of the derive macro `Properties` (in Nightly builds, run with -Z macro-backtrace for more info)
